pkg/bpf/perf.go

```diff
@@ -443,6 +456,11 @@ func (e *PerfEvent) Close() {
 	unix.Close(e.Fd)
 }
 
+// Debug returns string with internal information about PerfEvent
+func (e *PerfEvent) Debug() string {
+	return fmt.Sprintf("cpu: %d, Fd: %d, pagesize: %d, npages: %d, lost: %d, unknown: %d, data: %v, state: %v, buf: %v", e.cpu, e.Fd, e.pagesize, e.npages, e.lost, e.unknown, e.data, C.GoBytes(e.state, C.sizeof_struct_read_state), e.buf)
```

Comment:
Do we have an example of what this looks like?

At a glance, I don't know how `e.state` will be rendered to text, which will be very important. If we have to decode this from raw bytes, it's doable but it would be nice to get a more readable version.

Aren't `e.data` and `e.buf` just raw buffers? Will this format them in some way or just print the raw pointers?

Reply:
```
Oct 10 22:05:43 k8s1 cilium-agent[4226]:  1 0 32 0 20 0 0 0 2 45 181 203 201 250 247 211 0 6 0 0 0 0 0 0 0 0 0 0 9 0 0 0 1 0 32 0 20 0 0 0 2 8 181 203 201 250 247 211 18 236 5 0 0 0 0 0 0 0 0 0 9 0 0 0 1 0 32 0 20 0 0 0 2 15 181 203 201 250 247 211 1 0 0 0 0 0 0 0 0 0 0 0 9 0 0 0 1 0 176 0 164 0 0 0 4 0 181 203 201 250 247 211 199 0 0 0 128 0 0 0 1 0 0 0 7 149 0 0 181 203 1 0 126 6 0 0 130 183 181 115 116 117 138 219 68 3 124 173 8 0 69 0 0 185 13 29 64 0 63 6 243 84 10 0 0 1 10 0 38 205 193 224 39 70 111 167 242 9 139 70 82 146 128 24 0 229 59 121 0 0 1 1 8 10 5 116 255 19 5 116 255 19 71 69 84 32 47 104 101 97 108 116 104 99 104 101 99 107 47 107 117 98 101 100 110 115 32 72 84 84 80 47 49 46 49 13 10 72 111 115 116 58 32 49 48 46 48 46 51 56 46 50 48 53 58 49 48 48 53 52 13 10 85 115 0 0 0 0 9 0 0 0 1 0 112 0 100 0 0 0 4 5 181 203 17 187 254 16 66 0 0 0 66 0 0 0 7 149 0 0 0 0 0 0 0 0 0 0 0 0 0 0 138 219 68 3 124 173 130 183 181 115 116 117 8 0 69 0 0 52 173 129 64 0 64 6 82 117 10 0 38 205 10 0 0 1 39 70 193 224 139 70 82 146 111 167 242 142 128 16 0 235 58 244 0 0 1 1 8 10 5 116 255 19 5 116 255 19 0 0 9 0 0 0 1 0 32 0 20 0 0 0 2 44 181 203 17 187 254 16 10 0 38 205 10 0 0 1 70 39 224 193 9 0 0 0 1 0 32 0 20 0 0 0 2 45 181 203 17 187 254 16 1 6 0 0 0 0 0 0 0 0 0 0 9 0 0 0 1 0 32 0 20 0 0 0 2 8 181 203 17 187 254 16 18 236 5 0 0 0 0 0 0 0 0 0 9 0 0 0 1 0 32 0 20 0 0 0 2 15 181 203 17 187 254 16 2 0 0 0 0 0 0 0 0 0 0 0 9 0 0 0 1 0 32 0 20 0 0 0 2 54 181 203 17 187 254 16 10 0 0 1 1 0 0 0 0 0 0 0 9 0 0 0 1 0 32 0 20 0 0 0 2 19 181 203 17 187 254 16 1 0 0 0 0 0 0 0 0 0 0 0 9 0 0 0 1 0 112 0 100 0 0 0 4 2 181 203 17 187 254 16 66 0 0 0 66 0 0 0 7 149 0 0 1 0 0 0 0 0 2 0 10 0 0 0 102 133 237 80 28 25 138 219 68 3 124 173 8 0 69 0 0 52 173 129 64 0 63 6 83 117 10 0 38 205 10 0 0 1 39 70 193 224 139 70 82 146 111 167 242 142 128 16 0 235 58 244 0 0 1 1 8 10 5 116 255 19 5 116 255 19 0 0 9 0 0 0 1 0 104 0 92 0 0 0 3 4 181 203 17 187 254 16 66 0 0 0 66 0 0 0 10 0 0 0 0 0 0 0 102 133 237 80 28 25 138 219 68 3 124 173 8 0 69 0 0 52 173 129 64 0 63 6 83 117 10 0 38 205 10 0 0 1 39 70 193 224 139 70 82 146 111 167 242 142 128 16 0 235 58 244 0 0 1 1 8 10 5 116 255 19 5 116 255 19 0 0 9 0 0 0 1 0 176 0 164 0 0 0 4 5 181 203 17 187 254 16 244 0 0 0 128 0 0 0 7 149 0 0 0 0 0 0 0 0 0 0 0 0 0 0 138 219 68 3 124 173 130 183 181 115 116 117 8 0 69 0 0 230 173 130 64 0 64 6 81 194 10 0 38 205 10 0 0 1 39 70 193 224 139 70 82 146 111 167 242 142 128 24 0 235 59 166 0 0 1 1 8 10 5 116 255 19 5 116 255 19 72 84 84 80 47 49 46 49 32 50 48 48 32 79 75 13 10 67 111 110 116 101 110 116 45 84 121 112 101 58 32 97 112 112 108 105 99 97 116 105 111 110 47 106 115 111 110 13 10 68 97 116 101 58 32 87 101 100 44 32 49 48 0 0 0 0 9 0 0 0 1 0 32 0 20 0 0 0 2 44 181 203 17 187 254 16 10 0 38 205 10 0 0 1 70 39 224 193 9 0 0 0 1 0 32 0 20 0 0 0 2 45 181 203 17 187 254 16 1 6 0 0 0 0 0 0 0 0 0 0 9 0 0 0 1 0 32 0 20 0 0 0 2 8 181 203 17 187 254 16 18 236 5 0 0 0 0 0 0 0 0 0 9 0 0 0 1 0 32 0 20 0 0 0 2 15 181 203 17 187 254 16 2 0 0 0 0 0 0 0 0 0 0 0 9 0 0 0 1 0 32 0 20 0 0 0 2 54 181 203 17 187 254 16 10 0 0 1 1 0 0 0 0 0 0 0 9 0 0 0 1 0 32 0 20 0 0 0 2 19 181 203 17 187 254 16 1 0 0 0 0 0 0 0 0 0 0 0 9 0 0 0 1 0 176 0 164 0 0 0 4 2 181 203 17 187 254 16 244 0 0 0 128 0 0 0 7 149 0 0 1 0 0 0 0 0 2 0 10 0 0 0 102 133 237 80 28 25 138 219 68 3 124 173 8 0 69 0 0 230 173 130 64 0 63 6 82 194 10 0 38 205 10 0 0 1 39 70 193 224 139 70 82 146 111 167 242 142 128 24 0 235 59 166 0 0 1 1 8 10 5 116 255 19 5 116 255 19 72 84 84 80 47 49 46 49 32 50 48 48 32 79 75 13 10 67 111 110 116 101 110 116 45 84 121 112 101 58 32 97 112 112 108 105 99 97 116 105 111 110 47 106 115 111 110 13 10 68 97 116 101 58 32 87 101 100 44 32 49 48 0 0 0 0 9 0 0 0 1 0 168 0 156 0 0 0 3 4 181 203 17 187 254 16 244 0 0 0 128 0 0 0 10 0 0 0 0 0 0 0 102 133 237 80 28 25 138 219 68 3 124 173 8 0 69 0 0 230 173 130 64 0 63 6 82 194 10 0 38 205 10 0 0 1 39 70 193 224 139 70 82 146 111 167 242 142 128 24 0 235 59 166 0 0 1 1 8 10 5 116 255 19 5 116 255 19 72 84 84 80 47 49 46 49 32 50 48 48 32 79 75 13 10 67 111 110 116 101 110 116 45 84 121 112 101 58 32 97 112 112 108 105 99 97 116 105 111 110 47 106 115 111 110 13 10 68 97 116 101 58 32 87 101 100 44 32 49 48 0 0 0 0 9 0 0 0 1 0 112 0 100 0 0 0 4 5 181 203 17 187 254 16 66 0 0 0 66 0 0 0 7 149 0 0 0 0 0 0 0 0 0 0 0 0 0 0 138 219 68 3 124 173 130 183 181 115 116 117 8 0 69 0 0 52 173 131 64 0 64 6 82 115 10 0 38 205 10 0 0 1 39 70 193 224 139 70 83 68 111 167 242 142 128 17 0 235 58 244 0 0 1 1 8 10 5 116 255 19 5 116 255 19 0 0 9 0 0 0 1 0 32 0 20 0 0 0 2 44 181 203 17 187 254 16 10 0 38 205 10 0 0 1 70 39 224 193 9 0 0 0 1 0 32 0 20 0 0 0 2 45 181 203 17 187 254 16 1 6 0 0 0 0 0 0 0 0 0 0 9 0 0 0 1 0 32 0 20 0 0 0 2 8 181 203 17 187 254 16 18 236 5 0 0 0 0 0 0 0 0 0 9 0 0 0 1 0 32 0 20 0 0 0 2 15 181 203 17 187 254 16 2 0 0 0 0 0 0 0 0 0 0 0 9 0 0 0 1 0 32 0 20 0 0 0 2 54 181 203 17 187 254 16 10 0 0 1 1 0 0 0 0 0 0 0 9 0 0 0 1 0 32 0 20 0 0 0 2 19 181 203 17 187 254 16 1 0 0 0 0 0 0 0 0 0 0 0 9 0 0 0 1 0 112 0 100 0 0 0 4 2 181 203 17 187 254 16 66 0 0 0 66 0 0 0 7 149 0 0 1 0 0 0 0 0 2 0 10 0 0 0 102 133 237 80 28 25 138 219 68 3 124 173 8 0 69 0 0 52 173 131 64 0 63 6 83 115 10 0 38 205 10 0 0 1 39 70 193 224 139 70 83 68 111 167 242 142 128 17 0 235 58 244
[...]
```